src/layout/List/ListComponent.tsx

```diff
@@ -55,50 +46,34 @@ export const ListComponent = ({
     (state) => state.dataListState.dataLists[id]?.paginationData?.totaltItemsCount || 0,
   );
 
-  const handleChange = ({ selectedValue }: ChangeProps) => {
+  const handleChange = ({ selectedValue: selectedValue }: ChangeProps<Record<string, string>>) => {
     for (const key in formData) {
       handleDataChange(selectedValue[key], { key: key });
     }
   };
 
-  const renderRow = (datalist) => {
-    const cells: JSX.Element[] = [];
-    for (const key of Object.keys(datalist)) {
-      cells.push(<TableCell key={`${key}_${datalist[key]}`}>{datalist[key]}</TableCell>);
-    }
-    return cells;
-  };
-
-  const renderHeaders = (headers) => {
-    const cell: JSX.Element[] = [];
-    for (const header of headers) {
-      if ((sortableColumns || []).includes(header)) {
-        cell.push(
-          <TableCell
-            onChange={handleSortChange}
-            sortKey={header}
-            key={header}
-            sortDirecton={sortColumn === header ? sortDirection : SortDirection.NotActive}
-          >
-            {getTextResourceAsString(header)}
-          </TableCell>,
-        );
-      } else {
-        cell.push(<TableCell key={header}>{getTextResourceAsString(header)}</TableCell>);
+  const selectedRow: Record<string, string> = React.useMemo(() => {
+    let matchRow: boolean[] = [];
+    for (const row of calculatedDataList) {
+      for (const key in formData) {
+        matchRow.push(formData[key] === row[key]);
+      }
+      if (!matchRow.includes(false)) {
+        return row;
       }
+      matchRow = [];
     }
-    return cell;
-  };
+    return {};
+  }, [formData, calculatedDataList]);
 
   const dispatch = useAppDispatch();
 
-  const handleSortChange = ({ sortedColumn, previousSortDirection }: SortProps) => {
+  const handleSortChange = (props: SortProps & { column: keyof Record<string, string> }) => {
```

Comment:
```suggestion
  const handleSortChange = (props: SortProps & { column: string }) => {
```
A "key in a record where the keys are strings", can be reduced to just use the "string" type. :smile: